src/app/drive/components/ShareDialog/ShareDialog.tsx

```diff
@@ -246,11 +246,16 @@ const ShareDialog = (props: ShareDialogProps): JSX.Element => {
 
     const isItemNotSharedYet = !isAdvancedShareItem(itemToShare?.item) && !itemToShare.item.sharings?.length;
 
+    const sharingInfo = await shareService.getSharingInfo(itemId, itemType).catch(() => {
+      return null;
+    });
+
+    sharingType = sharingInfo?.type ?? 'public';
+    isAlreadyPasswordProtected = sharingInfo?.publicSharing.isPasswordProtected ?? false;
+
     if (!isItemNotSharedYet) {
       try {
         const sharingData = await shareService.getSharingType(itemId, itemType);
-        sharingType = sharingData.type;
-        isAlreadyPasswordProtected = sharingData.encryptedPassword !== null;
         setSharingMeta(sharingData);
```

Comment:
What is the difference between `getSharingType` and `getSharingInfo` if we now get the type from the `getSharingInfo` call? Do we still need `getSharingType`?